src/NumericalAlgorithms/Interpolation/LinearInterpolationND.hpp

```diff
@@ -0,0 +1,374 @@
+                     CompressedIndexRecursion_t<CompressedIndex_t, 1>::value(
+                         indexArray, data.numberOfPoints)];
+
+      if constexpr (UniformSpacing) {
+        return linearInterpolation1D_Uniform(
+            A, B, xA, xB, data.inverseSpacing[whichDimension], x1);
+      } else {
+        return linearInterpolation1D_General(A, B, xA, xB, x1);
+      }
+    }
+  };
+
+  // Unpack in dimension!
+  template <typename Data_t, size_t whichDimension, typename CompressedIndex_t,
+            typename X1, typename... Xt>
+  struct RecursiveInterpolationKernelGeneral {
+    static inline double const interpolate(
+        Data_t const& data, std::array<size_t, Dimension> const& indexArray,
+        size_t which_var, Xt const&... xin, X1 const& x1) {
+      auto const xA = data.x[whichDimension][indexArray[whichDimension]];
+      auto const xB = data.x[whichDimension][indexArray[whichDimension] + 1];
+
+      auto const A = RecursiveInterpolationKernelGeneral<
+          Data_t, whichDimension - 1,
+          CompressedIndexRecursion_t<CompressedIndex_t, 0>,
+          Xt...>::interpolate(data, indexArray, which_var, xin...);
+      auto const B = RecursiveInterpolationKernelGeneral<
+          Data_t, whichDimension - 1,
+          CompressedIndexRecursion_t<CompressedIndex_t, 1>,
+          Xt...>::interpolate(data, indexArray, which_var, xin...);
+
+      if constexpr (UniformSpacing) {
+        return linearInterpolation1D_Uniform(
+            A, B, xA, xB, data.inverseSpacing[whichDimension], x1);
+      } else {
+        return linearInterpolation1D_General(A, B, xA, xB, x1);
+      }
+    }
+  };
+
+  // Unpack in dimension!
+  template <typename Data_t, typename CompressedIndex_t, typename X1>
+  struct RecursiveInterpolationKernelGeneral<Data_t, 0, CompressedIndex_t, X1> {
+    static inline double const interpolate(
+        Data_t const& data, std::array<size_t, Dimension> const& indexArray,
+        size_t which_var, X1 const& x1) {
+      size_t constexpr whichDimension = 0;
+
+      auto const xA = data.x[whichDimension][indexArray[whichDimension]];
+      auto const xB = data.x[whichDimension][indexArray[whichDimension] + 1];
+
+      auto const A =
+          data.y[which_var +
+                 NumberOfVariables *
+                     CompressedIndexRecursion_t<CompressedIndex_t, 0>::value(
+                         indexArray, data.numberOfPoints)];
+      auto const B =
+          data.y[which_var +
+                 NumberOfVariables *
+                     CompressedIndexRecursion_t<CompressedIndex_t, 1>::value(
+                         indexArray, data.numberOfPoints)];
+
+      if constexpr (UniformSpacing) {
+        return linearInterpolation1D_Uniform(
+            A, B, xA, xB, data.inverseSpacing[whichDimension], x1);
+      } else {
+        return linearInterpolation1D_General(A, B, xA, xB, x1);
+      }
+    }
+  };
+
+ public:
+  inline int getSize(size_t const whichDimension) {
```

Comment:
no need to mark anything inside a class definition as `inline`, it is implicitly.

we use "west const" (`const size_t`)

we use `size_t` for sizes (for better or worse...)

mark member function as `const` when possible

we typically call this `extents(const size_t dimension)`

Reply:
Thanks for the background! I changed things accordingly.